java-lib/src/main/java/com/wavefront/metrics/JsonMetricsGenerator.java

```diff
@@ -245,8 +288,18 @@ public static void writeRegularMetrics(Processor processor, JsonGenerator json,
           json.writeStartObject();
           json.writeFieldName("tags");
           json.writeStartObject();
-          for (Map.Entry<String, String> tagEntry : ((TaggedMetricName) subEntry.getKey()).getTags().entrySet()) {
-            json.writeStringField(tagEntry.getKey(), tagEntry.getValue());
+          if (subEntry.getKey() instanceof TaggedMetricName) {
+            for (Map.Entry<String, String> tagEntry : ((TaggedMetricName) subEntry.getKey()).getTags().entrySet()) {
+              json.writeStringField(tagEntry.getKey(), tagEntry.getValue());
+            }
+          }
+          if (pointTags != null) {
+            for (Map.Entry<String, String> tagEntry : pointTags.entrySet()) {
+              if (!(subEntry.getKey() instanceof TaggedMetricName) ||
```

Comment:
can you make 292 to 298 a little easier to read, by extracting the case where it's a TaggedMetricName instead of having that many instanceof/casts? thanks